gatsby-browser.js

```diff
@@ -1 +1,9 @@
 import '@reach/skip-nav/styles.css';
+
+// On the route update we can access the current path and add a state to each navigation, which we call prevLocation.
+// Defaults to the home route as thats where we start
+export const onRouteUpdate = ({ location, prevLocation }) => {
+  if (location && location.state) {
+    location.state.referrer = prevLocation ? prevLocation.pathname : '/';
```

Comment:
@molebox ah, yeah we're going to need to refactor this a bit. Even on a fresh page load `referrer` does not exist. Defaulting to 'null' is likely the right move here.

Reply:
Yep, good shout, it crossed my mind but I chose (incorrectly) to not go that route. I can fix it tomorrow if there is no rush, otherwise it's a pretty simple fix.